### PR TITLE
Prevent editing word lists for archived projects

### PR DESCRIPTION
**Pull request by cpeel**
Word lists are stored in the project directory and thus if the directory has been deleted (because the project was archived or deleted) the project's word lists cannot be managed. Prevent the link from showing up on the project page and do a more robust job of checking for these edgecases in the edit word lists page and failing early. This fixes an error seen in `php_errors` from PROD.

This consolidates some of the validation logic in `edit_project_word_lists.php`. It's still a hot mess (which I can say with both authority and shame since I wrote it all 14 years ago).

Testable in the [fix-archived-wordlist-edits](https://www.pgdp.org/~cpeel/c.branch/fix-archived-wordlist-edits/) sandbox.

Please help validate that this behaves like we would expect during project create. It should but that's the only real problematic edgecase that comes to mind.

### CODE REVIEW

File: tools/project_manager/edit_project_word_lists.php [hunk 1]
```diff
@@ -9,24 +9,22 @@
...
 include_once($relPath.'misc.inc');  // attr_safe(), html_safe()
-include_once($relPath.'faq.inc');  // attr_safe()
+include_once($relPath.'faq.inc');
 
 require_login();
 
 $return = array_get($_REQUEST, "return", "$code_url/tools/project_manager/projectmgr.php");
+$projectid = get_projectID_param($_REQUEST, 'projectid');
 
-if (!user_is_PM()) {
```
Comment (chrismiceli) [inline tools/project_manager/edit_project_word_lists.php:18]: This is now covered by user_can_edit_project now right?

**Review by chrismiceli — APPROVED** [review]

**Comment by srjfoo** [comment]
Everything looks good from the project page (pre-existing archived project and newly created empty project), but the link to edit word lists should probably be removed from the links in project search results See [pinc/ProjectSearchResults.inc#L332](https://github.com/DistributedProofreaders/dproofreaders/blob/43833ad204d81ec12d705737c50a6e42d4993559/pinc/ProjectSearchResults.inc#L332)

**Comment by cpeel** [comment]
> Everything looks good from the project page (pre-existing archived project and newly created empty project), but the link to edit word lists should probably be removed from the links in project search results See [pinc/ProjectSearchResults.inc#L332](https://github.com/DistributedProofreaders/dproofreaders/blob/43833ad204d81ec12d705737c50a6e42d4993559/pinc/ProjectSearchResults.inc#L332)

Ah, good catch. I've added a second commit that addresses it. The updated code removes the noncvs links which we'll add back in a PROD hack instead.

File: pinc/ProjectSearchResults.inc [hunk 2]
```diff
@@ -350,14 +351,15 @@ class OptionsColumn extends Column
             if ($project->state == PROJ_POST_FIRST_UNAVAILABLE ||
                 $project->state == PROJ_POST_FIRST_AVAILABLE ||
                 $project->state == PROJ_POST_FIRST_CHECKED_OUT) {
-                $actions .= "<a href=\"$projects_url/$projectid/$projectid.zip\">" . _("Download") . "</a><br>\n";
+                $actions[] = "<a href=\"$project->url/$projectid.zip\">" . _("Download") . "</a>";
             }
             if ($project->state == PROJ_POST_SECOND_CHECKED_OUT ||
                 $project->state == PROJ_POST_COMPLETE) {
-                $actions .= "<a href=\"$projects_url/$projectid/".$projectid."_second.zip\">" . _("Download") . "</a><br>\n";
+                $actions[] = "<a href=\"$project->url/".$projectid."_second.zip\">" . _("Download") . "</a>";
```
Comment (srjfoo) [inline pinc/ProjectSearchResults.inc:358]: *Very* picky, I know, but, spaces around the dots?

**Review by srjfoo — APPROVED** [review]
Looks good.

In the project search results Actions column, are the links to db-req and dp-format PROD hacks? If so, since you changed the structure of the code for the links, do those need to be updated, too?

**Comment by srjfoo** [comment]
> Looks good.
> 
> In the project search results Actions column, are the links to db-req and dp-format PROD hacks? If so, since you changed the structure of the code for the links, do those need to be updated, too?

<grin> -- you beat me to it. I didn't read your comment here until after I had already made that comment!

File: pinc/ProjectSearchResults.inc (hunk 2 above)
Comment (cpeel) [reply pinc/ProjectSearchResults.inc]: Fixed by integrating the variable into the string itself.

**Review by srjfoo — APPROVED** [review]

**Pull request merged**